kratos/solving_strategies/convergencecriterias/mixed_generic_criteria.h

```diff
@@ -493,7 +492,13 @@ class MixedGenericCriteria : public ConvergenceCriteria< TSparseSpace, TDenseSpa
                     dof_dx = TSparseSpace::GetValue(rDx, dof_id);
 
                     const auto &r_current_variable = it_dof->GetVariable();
-                    int var_local_key = mLocalKeyMap[r_current_variable.IsComponent() ? r_current_variable.GetSourceVariable().Key() : r_current_variable.Key()];
+                    const KeyType key = r_current_variable.IsComponent() ? r_current_variable.GetSourceVariable().Key() : r_current_variable.Key();
+                    auto key_find = mLocalKeyMap.find(key);
+                    if (key_find == mLocalKeyMap.end()) {
+                        KRATOS_ERROR << "MixedGenericCriteria: The key " << key << " was not found in mLocalKeyMap." <<
+                          "Make sure the input variables in constructor and the dofset variables match.\n";
+                    }
```

Comment:
```suggestion
                    KRATOS_ERROR_IF(key_find == mLocalKeyMap.end()) << "MixedGenericCriteria: The key " << key << " was not found in mLocalKeyMap for variable " << r_current_variable.Name() << ". Make sure the input variables in constructor and the dofset variables match." << std::endl;
```
I would flush here (`std::endl` == `\n` + flush), since this anyway crashes the execution and thus is not performance critical